os/binfmt/libelf/libelf_load.c

```diff
@@ -157,11 +155,9 @@ static void elf_elfsize(struct elf_loadinfo_s *loadinfo)
 
 	/* Save the allocation size */
 #ifdef CONFIG_APP_BINARY_SEPARATION
-	loadinfo->textsize = 1 << mpu_log2regionceil(0, textsize);
-	loadinfo->rosize = 1 << mpu_log2regionceil(0, rosize);
-#else
-	loadinfo->textsize = textsize;
+	loadinfo->rosize = rosize;
 #endif
+	loadinfo->textsize = textsize;
 	loadinfo->datasize = datasize;
```

Comment:
Don't you need to cover these two lines together with CONFIG_OPTIMIZE_APP_RELOAD_TIME?

Reply:
No. textsize and datasize values are used even without this config.